src/screens/Feed.js

```diff
@@ -1,124 +1,122 @@
-/* eslint-disable react-native/no-inline-styles */
 import React from 'react';
-import {View, StyleSheet, FlatList} from 'react-native';
+import {
+  View,
+  StyleSheet,
+  FlatList,
+  TouchableWithoutFeedback,
+  Text,
+  SafeAreaView,
+  Alert,
+} from 'react-native';
+
 import SettingsButton from './app/Components/SettingsButton/SettingsButton';
 import ProfileAvatar from './app/Components/ProfileAvatar/ProfileAvatar';
 import Post from './app/Components/Post/Post';
+import Swiper from 'react-native-swiper';
+import {Divider} from 'react-native-paper';
+import firestore from '@react-native-firebase/firestore';
+import * as ScreenshotDetector from 'react-native-screenshot-detect';
+
 export default class FeedScreen extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      swiperIndex: 0,
+      posts: [],
+      loading: false,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchPosts();
+
+    this.eventEmitter = ScreenshotDetector.subscribe(() => {
+      Alert.alert(
+        'The post with caption',
+        posts[this.state.swiperIndex].caption,
+        'was screenshotted',
+      );
+    });
+  }
+
+  componentWillUnmount() {
+    ScreenshotDetector.unsubscribe(this.eventEmitter);
   }
 
+  fetchPosts = () => {
+    const subscriber = firestore()
+      .collection('posts')
+      .onSnapshot(querySnapshot => {
+        const posts = [];
+
+        querySnapshot.forEach(documentSnapshot => {
+          posts.push({
+            ...documentSnapshot.data(),
+            key: documentSnapshot.id,
+          });
+        });
+
+        this.setState({
+          posts,
+          loading: false,
+        });
+      });
+  };
+
   // renders each post function. Each post is a card with a cover(the image) and content. Content for now has a caption with the username and 2 buttons
   renderPosts = element => {
     const {
-      img,
+      allowScreenshot,
       caption,
+      dateCreated,
+      hearts,
+      image,
+      key,
+      uid,
       username,
-      comments,
-      likes,
-      likers,
-      bookmarkers,
-    } = element.item;
-    const uid = 'dennyliang'; // this is the acc im logged on
-    const isLiked = likers[uid] === true ? true : false; // if i liked it
-    const isBookmarked = bookmarkers[uid] === true ? true : false;
+    } = element;
+    // const isLiked = likers[uid] ? true : false; // if i liked it
+    // const isBookmarked = bookmarkers[uid] ? true : false;
+
     return (
-      <View>
-        <View style={styles.topBarViewStyles}>
-          <ProfileAvatar username={username} img={img} />
-          <SettingsButton icon="horizontal" />
+      <TouchableWithoutFeedback
```

Comment:
Why are we wrapping the whole post component to go to the UserProfile? What if a user just wants to (zoom) into an image (a feature we can add in the future). Should we just wrap the Post Header?